src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/SpgProducerEeeCsvDataTests.cs

```diff
@@ -64,6 +65,192 @@ public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligat
             }
         }
 
+        [Fact]
+        public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligationType_As_B2C_WithAllCategories()
+        {
+            using (DatabaseWrapper db = new DatabaseWrapper())
+            {
+                //Arrange
+                ModelHelper helper = new ModelHelper(db.Model);
+                var scheme = helper.CreateScheme();
+                string approvalNumber = "WEE/TE0000ST/SCH";
+                scheme.ApprovalNumber = approvalNumber;
+                var memberUpload = helper.CreateSubmittedMemberUpload(scheme);
+                int complianceYear = DateTime.Now.Year;
+                memberUpload.ComplianceYear = complianceYear;
+                string pnr1 = "PRN123";
+                string pnr2 = "PRN456";
+                string b2bObligationType = EnumHelper.GetDisplayName(ObligationType.B2B);
+                string b2cObligationType = EnumHelper.GetDisplayName(ObligationType.B2C);
+
+                var producer1 = helper.CreateProducerAsCompany(memberUpload, pnr1);
+                var producer2 = helper.CreateProducerAsCompany(memberUpload, pnr2);
+
+                var dataReturnVersion1 = helper.CreateDataReturnVersion(scheme, complianceYear, 1);
+                var dataReturnVersion2 = helper.CreateDataReturnVersion(scheme, complianceYear, 2);
+
+                var categories = EnumHelper.GetValues(typeof(WeeeCategory));
+                var maxCategoryId = categories.Max(x => x.Key);
+                int quarter1Tonnage = 0;
+                int quarter2Tonnage = 0;
+
+                for (int i = 1; i <= maxCategoryId; i++)
+                {
+                    quarter1Tonnage = quarter1Tonnage + 10;
+                    quarter2Tonnage = quarter2Tonnage + 5;
+                    helper.CreateEeeOutputAmount(dataReturnVersion1, producer1.RegisteredProducer, b2cObligationType, i, quarter1Tonnage);
+                    helper.CreateEeeOutputAmount(dataReturnVersion2, producer1.RegisteredProducer, b2cObligationType, i, quarter2Tonnage);
+                }
+
+                helper.CreateEeeOutputAmount(dataReturnVersion1, producer2.RegisteredProducer, b2bObligationType, 2, 400);
```

Comment:
maybe add a comment "as well as adding all the B2C entries, add a B2B entry so we can ensure it's not returned"